resources/macos_user.rb

```diff
@@ -76,6 +77,24 @@ def user_already_exists?
       mode '0600'
     end
   end
+
+  if property_is_set?(:groups)
+    if groups.is_a? String
+      group groups do
+        action :create
+        members username
+        append true
+      end
+    else
+      groups.each do |g|
+        group g do
```

Comment:
What happens if the array contains a non-string? Maybe a number or a ruby object?

Reply:
We can throw some type-checking in there and enforce the usage of an array of strings. Though it might make sense to to have a group called 1, 2, 3, 4..